src/modules/esl-share/core/esl-share-button.ts

```diff
@@ -0,0 +1,101 @@
+import {ESLBaseElement} from '../../esl-base-element/core';
+import {attr, boolAttr, listen} from '../../esl-utils/decorators';
+import {ENTER, SPACE} from '../../esl-utils/dom/keys';
+import {ESLShareActionRegistry} from './esl-share-action-registry';
+
+import type {ESLShareList} from './esl-share-list';
+
+export interface ShareButtonConfig {
+  'action': string;
+  'id': string;
+  'icon': string;
+  'iconBackground': string;
+  'link': string;
+  'name': string;
+  'title': string;
+}
+
+export class ESLShareButton extends ESLBaseElement {
+  public static is = 'esl-share-button';
+
+  public static DEFAULT_ICON_BG_COLOR: string = '#FFF';
+
+  @attr() public action: string;
+  @attr() public buttonId: string;
+  @attr() public link: string;
+  @attr() public name: string;
+  @boolAttr() public unavailable: boolean;
+
+  public static build(cfg: ShareButtonConfig): ESLShareButton | null {
+    const shareAction = ESLShareActionRegistry.instance.get(cfg.action);
+    if (!shareAction) return null;
+
+    const {isAvailable} = shareAction;
+    const $button = ESLShareButton.create();
+    $button.$$attr('action', cfg.action);
+    $button.$$attr('button-id', cfg.id);
+    $button.$$attr('link', cfg.link);
+    $button.$$attr('name', cfg.name);
+    $button.$$attr('aria-label', cfg.title);
+    $button.$$attr('unavailable', !isAvailable);
+    const $icon = document.createElement('span');
+    $icon.title = cfg.title;
+    $icon.classList.add('esl-share-icon');
+    $icon.innerHTML = cfg.icon;
+    $icon.setAttribute('style', `background-color:${cfg.iconBackground || ESLShareButton.DEFAULT_ICON_BG_COLOR};`);
+    $button.appendChild($icon);
+    return $button;
+  }
```

Comment:
Looks not right. 
 1) to many actions
 2) does not use the ability of the accessors (I hope we can convert it to `Object.assign($btn, cfg)`)
 3) We need to think about content building in the separate function